content/en/blog/_posts/2019-12-09-csi-volume-migration-beta.md

```diff
@@ -0,0 +1,149 @@
+How to try out CSI migration for existing plugins?
+If you are Kubernetes distributor that deploys in one of the environments listed below, now would be a good time to start testing the CSI migration and figuring out how to deploy/manage the appropriate CSI driver.
+
+To try out CSI migration in beta for an existing plugin you must be using Kubernetes v1.17 or higher. First, you must update/create a Kubernetes cluster with the feature flags `CSIMigration` (on by default in 1.17) and `CSIMigration{provider}` (off by default) enabled on the `kube-apiserver`, `kube-controller-manager`, as well as the `kubelet`’s. Where {provider} is the in-tree cloud provider storage type that is used in your cluster.
+
+You must also install the requisite CSI driver on your cluster - instructions for this can generally be found from you provider of choice. CSI migration is available for GCE Persistent Disk and AWS Elastic Block Store in beta as well as for Azure Ffile/Ddisk and Openstack /Cinder in alpha. Kubernetes distributors should look at automating the deployment and management (upgrade, downgrade, etc.) of the CSI Drivers they will depend on.
+
+To verify the feature flag is enabled and driver installed on a particular node you can get the CSI Node object. You should see the in-tree plugin name of the migrated plugin as well as your driver in the drivers list.
+
+```shell
+kubectl get csinodes -o yaml
+```
+
+```yaml
+- apiVersion: storage.k8s.io/v1
+  kind: CSINode
+  metadata:
+    annotations:
+      storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/gce-pd
+    name: test-node
+    ...
+  spec:
+    drivers:
+      name: pd.csi.storage.gke.io
+      ...
+```
+
+After the above set up is complete you can confirm that your cluster has functioning CSI migration by deploying a stateful workload using the legacy APIs.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-disk
+spec:
+  storageClassName: standard
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: web-server
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - mountPath: /var/lib/www/html
+         name: mypvc
+  volumes:
+   - name: mypvc
+     persistentVolumeClaim:
+       claimName: test-disk
+```
+
+Inspect that the pod is RUNNING after some time
+
+```shell
+kubectl get pods web-server
+
+NAME         READY   STATUS    RESTARTS   AGE
+web-server   1/1     Running   0          39s
+```
+
+To confirm that the CSI driver is actually serving your requests it may be prudent to check the container logs of the CSI Driver after exercising the storage management operations. Note that your container logs may look different depending on the provider used.
+
+```shell
+kubectl logs {CSIdriverPodName} --container={CSIdriverContainerName}
+
```

Comment:
It's best to separate code from sample output (replace this one blank line with two lines, both containing three backticks).

Note: there's no way to put this in as a GitHub suggestion due to the way the feature works.